pdr_backend/subgraph/subgraph_subscriptions.py

```diff
@@ -58,79 +58,71 @@ def fetch_filtered_subscriptions(
         where_clause = f", where: {{timestamp_gt: {start_ts}, timestamp_lt: {end_ts}}}"
 
     # pylint: disable=line-too-long
-    while True:
-        query = f"""
-            {{
-                predictSubscriptions(skip: {offset}, first: {chunk_size} {where_clause}) {{
+    query = f"""
+        {{
+            predictSubscriptions(skip: {skip}, first: {first} {where_clause}) {{
+                id
+                txId
+                timestamp
+                user {{
                     id
-                    txId
-                    timestamp
-                    user {{
-                        id
-                    }}
-                    predictContract {{
+                }}
+                predictContract {{
+                    id
+                    token {{
                         id
-                        token {{
-                            id
-                            name
-                            lastPriceValue
-                            nft{{
-                                nftData {{
-                                key
-                                value
-                                }}
+                        name
+                        lastPriceValue
+                        nft{{
+                            nftData {{
+                            key
+                            value
                             }}
                         }}
                     }}
                 }}
-            }}"""
-
-        logger.info("Querying subgraph... %s", query)
-        result = query_subgraph(
-            get_subgraph_url(network),
-            query,
-            timeout=20.0,
+            }}
+        }}"""
+
+    logger.info("Querying subgraph... %s", query)
+    result = query_subgraph(
+        get_subgraph_url(network),
+        query,
+        timeout=20.0,
+    )
+
+    if "data" not in result or not result["data"]:
+        return []
+
+    data = result["data"].get("predictSubscriptions", [])
+    if len(data) == 0:
+        return []
+
+    for subscription_sg_dict in data:
+        info725 = subscription_sg_dict["predictContract"]["token"]["nft"]["nftData"]
+        info = info725_to_info(info725)
+        pair = info["pair"]
+        timeframe = info["timeframe"]
+        source = info["source"]
+        timestamp = subscription_sg_dict["timestamp"]
+        tx_id = subscription_sg_dict["txId"]
+        last_price_value = (
+            float(subscription_sg_dict["predictContract"]["token"]["lastPriceValue"])
+            * 1.201
```

Comment:
I see it is not added by you but I don't know what means this value

Reply:
I'm not sure either..